src/senselab/audio/tasks/preprocessing.py

```diff
@@ -30,13 +27,93 @@ def resample_audio_dataset(audios: List[Audio], resample_rate: int, rolloff: flo
     """
     resampled_audios = []
     for audio in audios:
-        resampled = F.resample(audio.audio_data, audio.sampling_rate, resample_rate, rolloff=rolloff)
+        resampled = F.resample(audio.waveform, audio.sampling_rate, resample_rate, rolloff=rolloff)
         resampled_audios.append(
+            Audio(waveform=resampled, sampling_rate=resample_rate, metadata=audio.metadata, path_or_id=audio.path_or_id)
+        )
+    return resampled_audios
+
+
+def downmix_audios_to_mono(audios: List[Audio]) -> List[Audio]:
+    """Downmixes a list of Audio objects to mono by averaging all channels.
+
+    Args:
+        audios (List[Audio]): A list of Audio objects with a tensor representing the audio waveform.
+                                 Shape: (num_channels, num_samples).
+
+    Returns:
+        List[Audio]: The list of audio objects with a mono waveform averaged from all channels. Shape: (num_samples).
+    """
+    down_mixed_audios = []
+    for audio in audios:
+        if audio.waveform.dim() != 2 or audio.waveform.size(0) < 1:
+            raise ValueError("waveform should have shape (num_channels, num_samples)")
+
+        down_mixed_audio = audio.copy()
+        down_mixed_audio.waveform = audio.waveform.mean(dim=0)
```

Comment:
I worry slightly about this but maybe your test cases handle this, but does this guarantee that waveform is shape (1, num_samples)? I'm not 100% sure that the field_validator runs in this case but if it does then this is fine. If it doesn't, then use `audio.waveform.mean(dim=0, keepdim=True)`

Reply:
it was managed anyway, but i have now added `keepdim=True`  to make it clearer